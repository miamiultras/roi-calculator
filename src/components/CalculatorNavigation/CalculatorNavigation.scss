.calculator-navigation {
  margin: 20px 0;
  .element {
    width: 80px;
    height: 4px;
    display: inline-block;
    background: #59595b;
    margin: 16px 8px 0 8px; 
  }
  .active {
    background: #5db98f;
  }
}

@media only screen and (max-width: 432px) {
  .calculator-navigation {
    .element {
      width: 40px;
    }
  }
}
