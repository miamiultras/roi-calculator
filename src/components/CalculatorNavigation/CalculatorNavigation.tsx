import React from 'react';

import './CalculatorNavigation.scss';

const CalculatorNavigation = ({ step }: { step: number }) => (
  <div className='calculator-navigation'>
    <div className={`element ${step >= 1 && 'active'}`} />
    <div className={`element ${step >= 2 && 'active'}`} />
    <div className={`element ${step >= 3 && 'active'}`} />
  </div>
);

export default CalculatorNavigation;
