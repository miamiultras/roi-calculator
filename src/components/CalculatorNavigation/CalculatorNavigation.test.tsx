import * as React from 'react';
import { shallow } from 'enzyme';
import CalculatorNavigation from './CalculatorNavigation';

function setup() {
  const props = {};
  const wrapper = shallow(<CalculatorNavigation step={1} />);
  return {
    wrapper,
    props,
  };
}

describe('CalculatorNavigation component', () => {
  it('should match snapshot if no change was intended', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });
});
