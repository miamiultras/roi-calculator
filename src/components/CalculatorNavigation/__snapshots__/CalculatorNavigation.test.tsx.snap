// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalculatorNavigation component should match snapshot if no change was intended 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CalculatorNavigation
    step={1}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="element active"
        />,
        <div
          className="element false"
        />,
        <div
          className="element false"
        />,
      ],
      "className": "calculator-navigation",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "element active",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "element false",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "element false",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="element active"
          />,
          <div
            className="element false"
          />,
          <div
            className="element false"
          />,
        ],
        "className": "calculator-navigation",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "element active",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "element false",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "element false",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
